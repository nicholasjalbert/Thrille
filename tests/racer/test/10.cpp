#include <pthread.h>
#include <iostream>
#include <stdlib.h>
#include <unistd.h>
#include <stdio.h>

using namespace std;

int x;
bool y;
pthread_t child, child2;
pthread_mutex_t mutex, mutex2;
pthread_cond_t cond, cond2;

void setup() {
    pthread_mutex_init(&mutex, NULL);
    pthread_mutex_init(&mutex2, NULL);
    pthread_cond_init(&cond, NULL);
    pthread_cond_init(&cond2, NULL);
    y = false;
}

void teardown() {
    pthread_mutex_destroy(&mutex);
    pthread_mutex_destroy(&mutex2);
    pthread_cond_destroy(&cond);
    pthread_cond_destroy(&cond2);
}

void * routine1(void * arg) {
    x = 3;
}

void * routine2(void * arg) {
}

int main(int argc, char *argv[]) {
    setup();
    x = 2;
    pthread_create(&child, NULL, routine1, NULL);
    pthread_join(child, NULL);
    teardown();
}

