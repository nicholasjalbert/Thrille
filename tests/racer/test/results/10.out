No races discovered
Race Events: 17
